akka-persistence-testkit/src/main/scala/akka/persistence/testkit/javadsl/EventSourcedBehaviorTestKit.scala

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.testkit.javadsl
+
+import java.util.function.{ Function => JFunction }
+import java.util.{ List => JList }
+
+import scala.reflect.ClassTag
+
+import com.typesafe.config.Config
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Behavior
+import akka.annotation.ApiMayChange
+import akka.annotation.DoNotInherit
+import akka.persistence.testkit.scaladsl
+import akka.util.ccompat.JavaConverters._
+
+/**
+ * Testing of [[akka.persistence.typed.javadsl.EventSourcedBehavior]] implementations.
+ * It supports running one command at a time and you can assert that the synchronously returned result is as expected.
+ * The result contains the events emitted by the command and the new state after applying the events.
+ * It also has support for verifying the reply to an command.
+ *
+ * Serialization of commands, events and state are verified automatically.
+ */
+@ApiMayChange
+object EventSourcedBehaviorTestKit {
+
+  /**
+   * The configuration to be included in the configuration of the `ActorSystem`. Typically used as
+   * constructor parameter to `TestKitJunitResource`. The configuration enables the in-memory
+   * journal and snapshot storage.
+   */
+  val config: Config = scaladsl.EventSourcedBehaviorTestKit.config
+
+  val enabledSerializationSettings: SerializationSettings = new SerializationSettings(
+    enabled = true,
+    verifyEquality = false,
+    verifyCommands = true,
+    verifyEvents = true,
+    verifyState = true)
+
+  val disabledSerializationSettings: SerializationSettings = new SerializationSettings(
+    enabled = false,
+    verifyEquality = false,
+    verifyCommands = false,
+    verifyEvents = false,
+    verifyState = false)
+
+  /**
+   * Customization of which serialization checks that are performed.
+   * `equals` must be implemented (or using `case class`) when `verifyEquality` is enabled.
+   */
+  final class SerializationSettings(
+      val enabled: Boolean,
+      val verifyEquality: Boolean,
+      val verifyCommands: Boolean,
+      val verifyEvents: Boolean,
+      val verifyState: Boolean) {
```

Comment:
This is good, since not all use cases use sharding etc which means commands/replies don't always need serialization, and not all persistence stores use serialization to persist objects etc.